src/gst.rs

```diff
@@ -84,21 +98,47 @@ impl RtspServer {
         Ok(maybe_app_src)
     }
 
-    pub fn set_credentials(&mut self, user_pass: Option<(&str, &str)>) -> Result<()> {
-        let auth = user_pass.map(|(user, pass)| {
```

Comment:
I like the original function signature with user/pass being passed in together. That way you know the username/password both exist or neither exist when you arrive in the function, and validating the combo stays in main or config parsing or wherever.

Also, if you have an unspecified user or pass in the config, you'd want it to be `None`, not `""`. So change the config value to `Option<String>`, which should let you get rid of the associated default method.

In other words, keep the boundary between "stringly typed" config and strongly-typed Rust at the first opportunity.

Reply:
I did like your original too but I had trouble getting it to compile.

How would I get
```
let auth = user_pass.map(|(user, pass)| {
```
to work with the case of None
I still need the `auth` variable to set the default of allowing access.

If I don't let auth and instead do
```
user_pass.map(|(user, pass)| {
// stuff with pass
}
```
I still need to handle the none to setup default persmission

This is looking like it just needs a match but perhaps you have a better idea?

Reply:
You'd still get-or-create `auth` from the server object as you do now. After that, you'd optionally set the user/pass up if they exist:
```rust
if let Some((user, pass)) = user_pass {
    // Now user and pass are in scope, modify auth
} else {
    // They don't exist (user_pass == None) so do the default
}

Reply:
Ok, got it working but not quite the same format.. please comment

Reply:
Oh that's cleaner I will change it to the method you suggested

Reply:
Yeah you get this gist, `if let` is sugar over `match`.